### PR TITLE
Address recommendations from supply chain GHA audit

### PR DESCRIPTION
**Pull request by tcp13equals2**
This PR addresses recommendations raised from a GitHub actions supply chain audit. 

Changes by file

**.github/workflows/test-build-deploy.yml** — addresses the test job holding pull-requests: write while running untrusted PR code (go test), which would let a malicious PR test tamper with PR thread state via $GITHUB_TOKEN.
- Dropped pull-requests: write from the test matrix job; permissions are now contents: read only.
- Removed the Install GitHub CLI step (no longer needed in the test job).
- Each shard that fails now writes FAILED_PACKAGES to a small file and uploads it as a failed-packages-<tags>-<id> artifact.
- New downstream job notify-streamingpromql-failures (needs: test, permissions: pull-requests: write) downloads those artifacts, de-duplicates the failed-package list across shards, and posts a single PR comment if pkg/streamingpromql failed. Uses GH_TOKEN/GH_REPO env so gh pr comment works on a runner with no checkout.

**.github/workflows/test-docs.yml** — addresses the mutable grafana/docs-base:latest tag in a PR-triggered workflow.
- Pinned to grafana/docs-base@sha256:b4d886449a18b7b7532214b58ed08332cc76b9469924cb08bc0c71fe161dc950 (current latest digest, fetched from Docker Hub). Renovate's docker manager will keep it updated.

**.github/workflows/update-vendored-mimir-prometheus.yml** — addresses template injection via direct ${{ steps.*.outputs.* }} interpolation into a shell run: block, plus a missing persist-credentials: false.
- Replaced the direct ${{ … }} interpolation in the branch-names (combined) step with env: indirection (DISPATCH_MIMIR_BRANCH, MANUAL_MIMIR_BRANCH, etc.).
- Added persist-credentials: false to the checkout.

**.github/workflows/push-mimir-build-image.yml** — addresses persisting the App token globally via git config --global url.https://x-access-token:$TOKEN@github.com/.insteadOf, which leaves the token recoverable from ~/.gitconfig by any later step in the same runner.
- Replaced the global config rewrite with a one-shot git -c http.https://github.com/.extraheader=Authorization: Basic <base64> flag on the push invocation only.
- Tidied surrounding script: set -euo pipefail, early exit 0 instead of nested else, and lifted ${{ steps.compute_hash.outputs.tag }} into the existing $TAG env var.

**.github/workflows/generate-docs-helm-tests-renovate-pr.yml** — addresses third-party (personal-account) action ksivamuthu/actions-setup-gh-cli in a privileged job that holds the mimir-github-bot App token.
- Swapped for the in-repo ./.github/workflows/scripts/install-gh.sh (build image is Debian-based, so apt install works).

**.github/workflows/changelog-check.yml** — addresses missing persist-credentials: false on a PR-triggered checkout.
- Added persist-credentials: false.

**.github/workflows/flaky-tests.yml** — same as above, on a scheduled workflow that holds Vault secrets.
- Added persist-credentials: false.

**CODEOWNERS** — addresses stale entries referencing workflow files that don't exist (deploy-pr-preview.yml, update-make-docs.yml).
- Removed the two dead entries.
- Added the two real preview workflow files (deploy-pr-preview-mimir.yml, deploy-pr-preview-helm-charts.yml) under the same docs-tooling ownership.

Behaviour changes worth calling out in the PR description

- Streamingpromql-failure comments are now aggregated into one PR comment per workflow run, rather than one per matrix shard.
- gh CLI is no longer installed in the unit-test container (the deleted notify step was the only consumer).

Validation done

- All 7 changed YAML files parse cleanly (ruby -ryaml).
- The two non-trivial new shell blocks pass shellcheck (the two SC2016 hits on markdown backticks are intentional).
- A second-pass Codex review reported no findings.

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`. If changelog entry is not needed, please add the `changelog-not-needed` label to the PR.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

---

> [!NOTE]
> **Medium Risk**
> Changes CI permissions and notification behavior (aggregated streamingpromql comments); misconfiguration could break PR comments or docs/build workflows, but production Mimir is unaffected.
> 
> **Overview**
> This PR hardens **GitHub Actions** workflows after a supply-chain audit, without changing application runtime behavior.
> 
> The **unit test** matrix job no longer has `pull-requests: write` while running untrusted `go test` code; it only uploads failed-package artifacts. A new **`notify-streamingpromql-failures`** job (read-only checkout, `pull-requests: write`) aggregates those artifacts and posts **one** PR comment when `pkg/streamingpromql` fails on vendored-prometheus update PRs.
> 
> Several workflows add **`persist-credentials: false`** on checkout, pin **`grafana/docs-base`** to a digest in docs CI (with Renovate to bump it), replace a third-party **gh** install action with an in-repo script, avoid persisting the **App token** in global git config on build-image pushes (one-shot `extraheader` instead), and use **env indirection** for branch names in the vendoring workflow. **CODEOWNERS** is updated for real docs preview workflows; **Renovate** gains rules for `docs-base` digests and `grafana/shared-workflows` action tags.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 8c85c0a939873af4d8cd003fdf1ad2ea9da1ae5a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by karlskewes — DISMISSED** [review]

**Comment by karlskewes** [comment]
Please backport to active release branches: 3.1, 3.0, 2.17.

**Review by karlskewes — DISMISSED** [review]

File: renovate.json5 [hunk 1]
```diff
@@ -24,6 +24,20 @@
...
+      // This custom manager tracks the digest against the :latest tag so it
+      // can be auto-bumped via a PR rather than baked in at commit time.
+      customType: 'regex',
+      description: 'Update grafana/docs-base digest in workflow files',
+      managerFilePatterns: ['.github/workflows/**'],
+      matchStrings: [
+        '(?<depName>grafana/docs-base)@(?<currentDigest>sha256:[a-f0-9]+)',
+      ],
+      currentValueTemplate: 'latest',
+      datasourceTemplate: 'docker',
```
Comment (karlskewes) [inline renovate.json5:39]: I think this comes from here: https://docs.renovatebot.com/modules/datasource/docker/#description

Comment (karlskewes) [reply renovate.json5:39]: And https://docs.renovatebot.com/modules/manager/regex/#configuration-templates

**Review by karlskewes — DISMISSED** [review]

File: .github/workflows/test-build-deploy.yml [hunk 2]
```diff
@@ -342,24 +342,65 @@ jobs:
...
+    permissions:
+      contents: read
+      pull-requests: write
+    steps:
+      - name: Download failed packages artifacts
+        uses: actions/download-artifact@3e5f45b2cfb9172054b4087a40e8e0b5a5461e7c # v8
+        with:
+          pattern: failed-packages-*
+          merge-multiple: true
+          path: failed-packages
```
[the remark this answers is not in the document]
Comment (tcp13equals2) [reply .github/workflows/test-build-deploy.yml:377]: https://github.com/grafana/mimir/pull/15424/changes/8c85c0a939873af4d8cd003fdf1ad2ea9da1ae5a

**Review by karlskewes — APPROVED** [review]
